lib/controllers/login.js

```diff
@@ -4,152 +4,168 @@ var _ = require('lodash');
-  var nextUri = url.parse(req.query.next || '').path;
-  var formActionUri = (config.web.login.uri + (nextUri ? ('?next=' + nextUri) : ''));
-  var view = req.app.get('stormpathConfig').web.login.view;
-  var oauthStateToken = null;
-  var providerData = req.body && req.body.providerData;
-
-  function renderForm(form, options) {
-    if (options === undefined) {
-      options = {};
-    }
-
-    var hasSocialProviders = _.some(config.socialProviders, function (socialProvider) {
-      return socialProvider.enabled;
-    });
 
-    extend(options, {
-      form: form,
-      formActionUri: formActionUri,
-      oauthStateToken: oauthStateToken,
-      hasSocialProviders: hasSocialProviders
-    });
+  credentials = credentials || {};
+  credentials.username = credentials.login || credentials.username || '';
+  credentials.password = credentials.password || '';
 
-    helpers.render(req, res, view, options);
+  if (!credentials.username || !credentials.password) {
+    return callback(new Error('Invalid username or password.'));
   }
 
-  // If a GET request is made with the Accept header as json,
-  // respond with the view model for this form.
-  // TODO: When `stormpath.web.produces` is implemented,
-  // this `if` must check that it is set to `application/json`.
-  if (req.method === 'GET' && accepts === 'json') {
-    return helpers.getLoginViewModel(config, function (err, viewModel) {
+  authenticator.authenticate(credentials, function (err, authenticationResult) {
+    if (err) {
+      return callback(err);
+    }
+
+    authenticationResult.getAccount(function (err, account) {
       if (err) {
-        return res.status(400).json({ error: err.userMessage || err.message });
+        return callback(err);
       }
 
-      res.status(200).json(viewModel);
+      callback(null, {
+        account: account,
+        authenticationResult: authenticationResult
+      });
     });
-  }
+  });
+}
 
-  if (req.user && config.web.login.enabled) {
-    var nextUrl = nextUri || config.web.login.nextUri;
-    return res.redirect(302, nextUrl);
-  }
+/**
+ * This controller logs in an existing user.  If there are any errors, an
+ * error page is rendered.  If the process succeeds, the user will be logged in
+ * and redirected.
+ *
+ * @method
+ *
+ * @param {Object} req - The HTTP request.
+ * @param {Object} res - The HTTP response.
+ * @param {function} next - The next function.
+ */
+module.exports = function (req, res, next) {
+  var config = req.app.get('stormpathConfig');
+  var application = req.app.get('stormpathApplication');
 
   res.locals.status = req.query.status;
 
-  if (req.method === 'POST' && accepts === 'json') {
-    if (providerData) {
-      // Social Login
-      var options = {
-        providerData: {
-          providerId: providerData.providerId,
-          accessToken: providerData.accessToken,
-          code: providerData.code
-        }
-      };
+  helpers.handleContentTypeRequest(req, {
```

Comment:
> The Content-Type header is used by the response. 

@timothyej that's not correct, the `Content-Type` header declares the format of the incoming POST body - as such this is related to the request, not the response.

Your other concern is valid, `handleAcceptRequest` would be more correct as the function is branching based on the Accept header.